### PR TITLE
Update Set-CsTenantFederationConfiguration.md

### PR DESCRIPTION
**Pull request by martinpechal1**
Adding new setting for TenantFederationConfiguration:

### -ExternalAccessWithTrialTenants
When set to 'Blocked', all external access with Teams subscriptions that are all trial will be blocked. Users from trial tenants will not be able to reach to users from tenants that have enabled the Block and vice-versa. The setting will also remove trial tenant users from existing chats if the block gets enabled.
The default value of the setting is Blocked.

Allowed - Communication with other tenants is allowed

Blocked - Communication with users in tenants that contain only trial users will be blocked.

```yaml
Type: ExternalAccessWithTrialTenantsType
Parameter Sets: (All)
Aliases:
Applicable: Microsoft Teams

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### CODE REVIEW

File: teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md [hunk 1]
```diff
@@ -167,6 +168,14 @@ When the BlockAllSubdomains is enabled, all users from all subdomains of all dom
 So, users from subdomain.contoso.com and subdomain.fabrikam.com will be blocked.
 Note: Users from subcontoso.com will not be blocked because it's a completely different domain rather than a subdomain of contoso.com.
 
+### -------------------------- Example 11 -------------------------
+```
+Set-CsTenantFederationConfiguration -ExternalAccessWithTrialTenants "Allowed"
+```
+
+Example 11 shows how the external access to be able to communicate with tenants that contain trial users can be allowed (default value is Blocked).
```
Comment (williamlooney) [inline teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md:176]: "Example 11 shows how the external access to be able to communicate with tenants that contain trial users can be allowed (default value is Blocked)."

should say the following to align with the text used in examples 10 and others:

"Example 11 shows how you can allow users to communicate with users in tenants that contain only trial users (default value is Blocked)."

File: teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md [hunk 2]
```diff
@@ -439,6 +448,27 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -ExternalAccessWithTrialTenants
+When set to 'Blocked', all external access with Teams subscriptions that are all trial will be blocked. Users from trial tenants will not be able to reach to users from tenants that have enabled the Block and vice-versa. The setting will also remove trial tenant users from existing chats if the block gets enabled.
+The default value of the setting is Blocked.
+
+Allowed - Communication with other tenants is allowed
+
+Blocked - Communication with users in tenants that contain only trial users will be blocked.
```
Comment (williamlooney) [inline teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md:457]: Communication with users in tenants that contain only trial licenses will be blocked

File: teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md [hunk 3]
```diff
@@ -439,6 +448,27 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -ExternalAccessWithTrialTenants
+When set to 'Blocked', all external access with Teams subscriptions that are all trial will be blocked. Users from trial tenants will not be able to reach to users from tenants that have enabled the Block and vice-versa. The setting will also remove trial tenant users from existing chats if the block gets enabled.
+The default value of the setting is Blocked.
+
+Allowed - Communication with other tenants is allowed
```
Comment (williamlooney) [inline teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md:455]: Allowed - Communication with other tenants is allowed based on other settings.

File: teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md [hunk 4]
```diff
@@ -439,6 +448,27 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -ExternalAccessWithTrialTenants
+When set to 'Blocked', all external access with Teams subscriptions that are all trial will be blocked. Users from trial tenants will not be able to reach to users from tenants that have enabled the Block and vice-versa. The setting will also remove trial tenant users from existing chats if the block gets enabled.
```
Comment (williamlooney) [inline teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md:452]: Should read:

When set to 'Blocked', all external access with users from Teams subscriptions that contain only trial licenses will be blocked. This means users from these trial-only tenants will not be able to reach to your users via chats, Teams calls, and meetings (using the users authenticated identity) and your users will not be able to reach users in these trial-only tenants. If this setting is set to "Blocked", users from the trial-only tenant will also be removed from existing chats.

**Review by williamlooney — CHANGES_REQUESTED** [review]
Please see comments for language changes. Trying to make sure the text is aligned with other text in the cmdlet and explain what the setting does

File: teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md (hunk 4 above)
Comment (martinpechal1) [reply teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md]: Updated.

File: teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md (hunk 3 above)
Comment (martinpechal1) [reply teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md]: Updated.

File: teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md (hunk 2 above)
Comment (martinpechal1) [reply teams/teams-ps/teams/Set-CsTenantFederationConfiguration.md]: Updated.

**Review by williamlooney — APPROVED** [review]
Approved by Mgmt Infra PM lead

**Comment by martinpechal1** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Pull request merged**